src/Features/Core/Portable/RemoveUnusedVariable/AbstractRemoveUnusedVariableCodeFixProvider.cs

```diff
@@ -34,13 +34,25 @@ internal abstract class AbstractRemoveUnusedVariableCodeFixProvider<TLocalDeclar
 
         protected abstract SeparatedSyntaxList<SyntaxNode> GetVariables(TLocalDeclarationStatement localDeclarationStatement);
 
+        protected abstract bool ShouldOfferFixForLocalDeclaration(ISyntaxFactsService syntaxFacts, TLocalDeclarationStatement node);
+
         internal sealed override CodeFixCategory CodeFixCategory => CodeFixCategory.CodeQuality;
 
-        public sealed override Task RegisterCodeFixesAsync(CodeFixContext context)
+        public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
         {
             var diagnostic = context.Diagnostics.First();
-            context.RegisterCodeFix(new MyCodeAction(c => FixAsync(context.Document, diagnostic, c)), diagnostic);
-            return Task.CompletedTask;
+
+            var document = context.Document;
+            var root = await document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+            var node = root.FindNode(diagnostic.Location.SourceSpan);
+
+            var syntaxFacts = document.GetRequiredLanguageService<ISyntaxFactsService>();
+            var localDeclaration = FindLocalDeclaration(node);
+
+            if (localDeclaration is null || ShouldOfferFixForLocalDeclaration(syntaxFacts, localDeclaration))
```

Comment:
out of curiosity, when are we removing when there is no local variable decl?

Reply:
A declaration in a for loop, or catch clause for example

Reply:
Ah. The get ancestor call is wrong then. You could have a catch variable in a llama inside a local decl initializer.

Reply:
Lost to the auto-merge, logged https://github.com/dotnet/roslyn/issues/56924 and will follow up.

Reply:
That's fine